partiql-spi/src/main/kotlin/org/partiql/spi/connector/ConnectorMetadata.kt

```diff
@@ -16,12 +16,28 @@ package org.partiql.spi.connector
 
 import org.partiql.spi.BindingPath
 import org.partiql.types.StaticType
+import org.partiql.types.function.FunctionSignature
 
 /**
  * Aids in retrieving relevant Catalog metadata for the purpose of planning and execution.
  */
 public interface ConnectorMetadata {
 
+    /**
+     * Scalar function signatures available via call syntax.
+     */
+    public val functions: List<FunctionSignature.Scalar>
+
+    /**
+     * Scalar function signatures available via operator or special form syntax.
+     */
+    public val operators: List<FunctionSignature.Scalar>
+
+    /**
+     * Aggregation function signatures.
+     */
+    public val aggregations: List<FunctionSignature.Aggregation>
+
```

Comment:
This was already added in Plugin. What's the diff?

I've envisioned that the gathering of functions via the SQL Path would make use of the Connector Metadata. However, it would be more like `getFunction(path)`. Though, since we don't have the concept of the SQL Path yet, the functions exposed on Plugin would suffice.

Reply:
The plugin returns signatures and implementations. This is signatures only and allows us to give the planner access to function signatures from connector metadata without having to implement SQL Path